src/dropbox-base.js

```diff
@@ -265,10 +316,85 @@ export class DropboxBase {
             status: res.status,
           };
         }
+
+        if (data.refresh_token) {
+          return {
+            accessToken: data.access_token,
+            refreshToken: data.refresh_token,
+            accessTokenExpiresAt: getTokenExpiresAt(data.expires_in),
+          };
+        }
         return data.access_token;
       });
   }
 
+  /**
+   *
+   * @returns {Promise<[*, *]>}
+   */
+  checkAndRefreshAccessToken() {
+    const canRefresh = this.getRefreshToken() && this.getClientId() && this.getClientSecret();
+    const needsRefresh = this.getAccessTokenExpiresAt() &&
+            (new Date(Date.now() + TokenExpirationBuffer)) >= this.getAccessTokenExpiresAt();
+    const needsToken = !this.getAccessToken();
+    if ((needsRefresh || needsToken) && canRefresh) {
```

Comment:
What if we need to refresh / need token, but we can't refresh? Should there be error handling here as well?

Reply:
If we need to but we can't we will just fail with an expired access token an let the application decide how to handle that rather than throwing here